apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,182 @@
+    }
+
+    if (req.body.__action === 'add_new_override') {
+      const { user_id, group_id } = await getUserOrGroupId({
+        course_instance_id: res.locals.course_instance.id,
+        assessment: res.locals.assessment,
+        uid: req.body.student_uid,
+        group_name: req.body.group_name,
+      });
+
+      
+      await runInTransactionAsync(async () => {
+        const inserted = await sqldb.queryAsync(sql.insert_assessment_access_policy, {
+          assessment_id: res.locals.assessment.id,
+          created_by: res.locals.authn_user.user_id,
+          credit: req.body.credit,
+          end_date: new Date(req.body.end_date),
+          group_name: req.body.group_name || null,
+          note: req.body.note || null,
+          start_date: new Date(req.body.start_date),
+          group_id: group_id || null,
+          user_id: user_id || null,
+        });
+        await insertAuditLog({
+          authn_user_id: res.locals.authn_user.user_id,
+          table_name: 'assessment_access_policies',
+          action: 'insert',
+          institution_id: res.locals.institution.id,
+          course_id: res.locals.course.id,
+          course_instance_id: res.locals.course_instance.id,
+          new_state: JSON.stringify(inserted.rows),
+          row_id: inserted.rows[0].id,
+        });
+      });
+
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'delete_override') {
+      
+      await runInTransactionAsync(async () => {
+        const deletedAccessPolicy = await sqldb.queryAsync(sql.delete_assessment_access_policy, {
+          assessment_id: res.locals.assessment.id,
+          unsafe_assessment_access_policies_id: req.body.policy_id,
+        });
+        if (deletedAccessPolicy.rows.length > 0) {
+          await insertAuditLog({
+            authn_user_id: res.locals.authn_user.user_id,
+            table_name: 'assessment_access_policies',
+            action: 'delete',
+            institution_id: res.locals.institution.id,
+            course_id: res.locals.course.id,
+            course_instance_id: res.locals.course_instance.id,
+            old_state: JSON.stringify(deletedAccessPolicy.rows),
+            row_id: deletedAccessPolicy.rows[0].id,
+          });
+        }
+      });
+      
+
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'edit_override') {
+      const { user_id, group_id } = await getUserOrGroupId({
+        course_instance_id: res.locals.course_instance.id,
+        assessment: res.locals.assessment,
+        uid: req.body.student_uid,
+        group_name: req.body.group_name,
+      });
+
+      const oldStateAccessPolicy = await sqldb.queryAsync(sql.select_assessment_access_policy, {
+        policy_id: req.body.policy_id,
+      });
+      
+      
+      await runInTransactionAsync(async () => {
+        const editAccessPolicy = await sqldb.queryAsync(sql.update_assessment_access_policy, {
```

Comment:
We should always be able to find this row by this point.

```suggestion
        const editAccessPolicy = await sqldb.queryOneRowAsync(sql.update_assessment_access_policy, {
```

Reply:
done in fe842679d48e106410884ef395d3ca3f52b03437